pycbc/io/hdf.py

```diff
@@ -893,3 +893,28 @@ def recursively_save_dict_contents_to_group(h5file, path, dic):
             recursively_save_dict_contents_to_group(h5file, path + key + '/', item)
         else:
             raise ValueError('Cannot save %s type'%type(item))
+
+def combine_and_copy(f, files, group):
+    """ Combine the same column from multiple files and save to a third"""
+    f[group] = np.concatenate([fi[group][:] if group in fi else \
+                                   np.array([], dtype=np.uint32) for fi in files])
+
+def name_all_datasets(files):
```

Comment:
Could these functions be simplified by using the `visit` / `visititems` methods in h5py? 
http://docs.h5py.org/en/latest/high/group.html#Group.visit 
(nb I just noticed myself that these existed ..)

Reply:
Have implemented this - it still needed some tweaking to be able to get the" dataset vs not dataset" test in it but it is now in, and probably much more efficient

Reply:
Trying to implement this, it's involving some variable-outside-function-within-function shuffling difficulty but hopefully should work

Reply:
This is now updated to use the h5py methods - however this is quite slow (taking around 90% of the running time of the script!) so should be looked at - though this is only a few seconds on a script which is not run many times

Reply:
I'm going to switch this back, I checked exactly the extra amount of time being used and there is logging on either side of this call:

With h5py methods:
`2019-02-06 08:54:48,405 generating list of datasets in input files
2019-02-06 08:54:55,371 copying foreground non-ifo-specific data`

Without h5py methods:
`2019-02-06 08:55:14,487 generating list of datasets in input files
2019-02-06 08:55:14,519 copying foreground non-ifo-specific data`

So an increase in this part of the code from around 30ms to 7s

Reply:
you mean much slower specifically using visit() and visititems() I guess?

Reply:
yes, sorry that's what i had meant (I specified this in the commit message that I've just pushed)